src/renderer/vt/vtrenderer.hpp

```diff
@@ -154,9 +154,18 @@ namespace Microsoft::Console::Render
         std::optional<TextColor> _newBottomLineBG{ std::nullopt };
 
         [[nodiscard]] HRESULT _Write(std::string_view const str) noexcept;
-        [[nodiscard]] HRESULT _WriteFormattedString(const std::string* const pFormat, ...) noexcept;
         [[nodiscard]] HRESULT _Flush() noexcept;
 
+        template<typename S, typename... Args>
+        [[nodiscard]] HRESULT _WriteFormatted(S&& format, Args&&... args)
+        try
+        {
+            fmt::basic_memory_buffer<char, 64> buf;
+            fmt::format_to(std::back_inserter(buf), std::forward<S>(format), std::forward<Args>(args)...);

```

Comment:
The `back_inserter` here isn't needed and I believe can result in performance degradation. `fmt` doesn't know how to append several characters at once when it's wrapped in a `back_insert_iterator`, and will fallback to stack allocating a 256 character buffer, fill that, and then copy it back character-per-character to `buf` instead of directly writing to it.

```suggestion
            fmt::format_to(buf, std::forward<S>(format), std::forward<Args>(args)...);
```

Reply:
@sylveon `fmt::format_to` is incompatible with the `FMT_COMPILE` macro that we use.
The next version of `fmt` will ship with `fmt::appender`, which we could then use instead of `std::back_inserter`. This upcoming version also removes the overloads for `fmt::basic_memory_buffer` anyways, forcing you to use either `fmt::appender` or `std::back_inserter`.

Reply:
Thanks @sylveon for the suggestion but sadly I failed to make it work, as described in https://github.com/microsoft/terminal/pull/10426#discussion_r654940971 . The current performance is amazingly good, almost as fast as hand-written formatting code. I'd be thrilled if we can go even further.

Reply:
Ah I must have missed that part then, my bad.